### PR TITLE
Refactor the plugins to use TypeScript

### PR DESCRIPTION
**Pull request by sonkkeli**

### CODE REVIEW

File: packages/shared/wbn-types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// These must match with the types in `wbn`.
+// github.com/WICG/webpackage/blob/main/js/bundle/src/constants.ts
+
+// TODO(sonkkeli): Should I rather export these from 'wbn' and then import?
```
Comment (cmfcmf) [inline packages/shared/wbn-types.ts:20]: I think it is fine like it is now :)

File: package.json [hunk 2]
```diff
@@ -1,14 +1,15 @@
 {
   "name": "webbundle-plugins",
   "scripts": {
-    "test": "ava",
+    "test": "rm -rf packages/lib-for-tests/ && tsc && ava && rm -rf packages/lib-for-tests/",
```
Comment (cmfcmf) [inline package.json:4]:
```suggestion
    "test": "rm -rf packages/lib-for-tests/ && tsc && ava; rm -rf packages/lib-for-tests/",
```

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
```
Comment (cmfcmf) [inline tsconfig.json:4]: Can you change this to 

```json
"lib": ["es2020"],
"module": "commonjs",
"target": "es2020",
```

to match what Node.js 14 supports? (https://www.npmjs.com/package/@tsconfig/node14)

File: packages/shared/iwa-headers.ts [hunk 4]
```diff
@@ -14,37 +14,39 @@
  * limitations under the License.
  */
 
-export const coep = Object.freeze({
+import { Headers, PluginOptions } from './types.js';
```
Comment (cmfcmf) [inline packages/shared/iwa-headers.ts:17]: This seems fishy to me.

File: package.json (hunk 2 above)
Comment (cmfcmf) [reply package.json:4]: Can you try 
NODE_PATH=./packages/lib-for-tests ava ...

File: tsconfig.json (hunk 3 above)
Comment (sonkkeli) [reply tsconfig.json]: We use modules instead of commonJs so changed to be es2020 to all 3 of them.

**Review by cmfcmf — APPROVED** [review]
Awesome work, thanks a lot!!

**Pull request merged**